src/components/Notification/Notification.test.tsx

```diff
@@ -57,3 +58,94 @@ describe('Inline Notification', () => {
     expect(closeCTA).toHaveBeenCalledTimes(1);
   });
 });
+
+describe('Notifications Container', () => {
+  const data1 = {
+    title: 'title1',
+    message: 'message1',
+    variant: 'banner' as NotificationVariants,
+    type: 'info' as NotificationTypes,
+    withIcon: true,
+    withFilling: false,
+    primaryCTALabel: 'primaryCTALabel1',
+  };
+
+  const data2 = {
+    title: 'title2',
+    message: 'message2',
+    variant: 'banner' as NotificationVariants,
+    type: 'info' as NotificationTypes,
+    withIcon: true,
+    withFilling: false,
+    primaryCTALabel: 'primaryCTALabel2',
+  };
+
+  test('Notifications Container with 2 Banner Notifications renders correctly', async () => {
+    const primaryCTA = jest.fn();
+    const closeCTA = jest.fn();
+
+    const { findByText } = render(
+      <NotificationsContainer position="top-right">
+        <Notification {...data1} primaryCTA={primaryCTA} closeCTA={closeCTA} />
+        <Notification {...data2} primaryCTA={primaryCTA} closeCTA={closeCTA} />
+      </NotificationsContainer>
+    );
+
+    const message1 = await findByText(data1.message);
+    expect(message1).toBeTruthy();
+
+    const message2 = await findByText(data2.message);
+    expect(message2).toBeTruthy();
+
+    const primaryCTALabel1 = await findByText(data1.primaryCTALabel);
+    expect(primaryCTALabel1).toBeTruthy();
+
+    const primaryCTALabel2 = await findByText(data1.primaryCTALabel);
+    expect(primaryCTALabel2).toBeTruthy();
+  });
+
+  test('Notifications Container Banner Notifications primaryCTAs work properly when clicked', async () => {
+    const primaryCTA1 = jest.fn();
```

Comment:
It's great to see tests here! Kudos Kostas!
Is there any specific reason for the double checks? 
I believe that you can just use one mocked function and one <Notification /> component to test all its functionalities.

Reply:
Well, given the fact that Notification Container can contain multiple Notification components I thought it would be nice to test that multiple primaryCTAs and closeCTAs work properly and aren't affecting each other (eg: each CTA click does not interfere with other CTAs etc). Well I'm not really sure if that's possible and if that's a necessary check; I'd like to hear your opinion on that approach.

Reply:
Your logic is correct, having many children components may affect things. Although, I think this has to do with UI things and not the functionality of the component.
Since the click event will be emitted inside the Notification component you don't have to worry if it will affect its siblings (this is an effect of event bubbling in JS).